runhouse/resources/hardware/cluster.py

```diff
@@ -1959,3 +1967,202 @@ def _folder_mv(
 
     def _folder_exists(self, path: Union[str, Path]):
         return self.client.folder_exists(path=path)
+
+    ###################################
+    # Cluster list helping methods
+    ###################################
+
+    @staticmethod
+    def _parse_time_duration(duration: str):
+        # A simple parser for duration like "15m", "2h", "3d"
+        unit = duration[-1]
+        value = int(duration[:-1])
+
+        # if the user provides a "--since" time filter than is less than a minute, set it by default to one minute.
+        if unit == "s" and value < 60:
+            value = 60
+
+        time_duration = value * TIME_UNITS.get(unit, 0)
+
+        if time_duration == 0 and value != 0:
+            logger.warning(
+                f"Filter is not applied, invalid time unit provided ({unit})."
+            )
+            return
+
+        return time_duration
+
+    @staticmethod
+    def _parse_filters(since: str, cluster_status: str):
+        cluster_filters = {}
+
+        if since:
+            last_active_in: int = Cluster._parse_time_duration(
```

Comment:
can we wrap in a try/except, so in the event an invalid value is passed in we just use the default?